deps/checksums/JuliaSyntax-25b737b1bbce259451f6d79997a25c541e7ab8c1.tar.gz/sha512

```diff
@@ -0,0 +1 @@
+13af8fb86e9286ce95c47811f5d37f18013a67ce4a3b1b8fdc97d283a8898f14584db2dc441519f4c645e20d622b8ffb9613f2fc141e4622b26f362f17a8c2f7
```

Comment:
CI error: https://buildkite.com/julialang/julia-master/builds/14799#0182ab15-7f58-4b9e-b57a-56addf3bed4d/552-2397

```
  ERROR: sha512 checksum failure on JuliaSyntax-25b737b1bbce259451f6d79997a25c541e7ab8c1.tar.gz, should be:
      13af8fb86e9286ce95c47811f5d37f18013a67ce4a3b1b8fdc97d283a8898f14
      584db2dc441519f4c645e20d622b8ffb9613f2fc141e4622b26f362f17a8c2f7
  But `sha512sum /cache/build/default-amdci5-1/julialang/julia-master/stdlib/srccache/JuliaSyntax-25b737b1bbce259451f6d79997a25c541e7ab8c1.tar.gz | awk '{ print $1; }'` results in:
      762532fcf6ec954165095e1892901d8afeff7b11fb4997aac2adbe0a7d572180
      d0eb49d1fa66edbf2559154206726cedd3c9db3362415e29836e1031897c6bcc
  This can happen due to bad downloads or network proxies, please check your
  network proxy/firewall settings and delete
  /cache/build/default-amdci5-1/julialang/julia-master/stdlib/srccache/JuliaSyntax-25b737b1bbce259451f6d79997a25c541e7ab8c1.tar.gz
  to force a redownload when you are ready
```

local test:

```sh
woclass@wos-PC:/mnt/v/tmp$ curl https://api.github.com/repos/JuliaLang/JuliaSyntax.jl/tarball/25b737b1bbce259451f6d79997a25c541e7ab8c1 -o JuliaSyntax.jl-25b737b1bbce259451f6d79997a25c541e7ab8c1.tar.gz
woclass@wos-PC:/mnt/v/tmp$ ls
JuliaSyntax.jl-25b737b1bbce259451f6d79997a25c541e7ab8c1.tar.gz  fasta.jl  fasta.txt  knucleotide.jl
woclass@wos-PC:/mnt/v/tmp$ sha512sum JuliaSyntax.jl-25b737b1bbce259451f6d79997a25c541e7ab8c1.tar.gz
cf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e  JuliaSyntax.jl-25b737b1bbce259451f6d79997a25c541e7ab8c1.tar.gz
```

Reply:
I get the checksum failure as well.

Reply:
The checksum `cf83e1357eefb8...` is just the checksum for the empty file, so something is weird with that `curl` command. It seems curl wasn't following the redirect for me, but adding `-L` didn't work.

I was able to download using `wget`:

```
wget https://api.github.com/repos/JuliaLang/JuliaSyntax.jl/tarball/25b737b1bbce259451f6d79997a25c541e7ab8c1
```

But then I was getting a different checksum each time, even though the archive content is the same.  Now when I run that exact same wget command nothing happens at all. So something *really* weird is going on here on the server side.

It's possible the weird behavior is due to me using the git tree hash rather than a commit hash in `JuliaSyntax.version`. Let me check/fix that.

Reply:
`cf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e` is the sha512 sum for an empty file :)

Anyway, so it seems this problem was due to using the tree hash rather than commit hash in `JuliaSyntax.version`.

Having done that, it seems github will do all sorts of weird stuff on the serverside, including
* Returning different `tgz` files each time but which unpack to the same tree
* Seemingly, randomly refusing to return a file at all and giving a 404 status instead

Reply:
That was confusing to debug.  Should be fixed now I hope.

Reply:
Ok, it works for me.

```sh
woclass@wos-PC:~$ wget -O JuliaSyntax-d3d8fdeaea37bcae05191cad50a326e608035d97.tar.gz  https://api.github.com/repos/JuliaLang/JuliaSyntax.jl/tarball/d3d8fdeaea37bcae05191cad50a326e608035d97
--2022-08-19 15:44:47--  https://api.github.com/repos/JuliaLang/JuliaSyntax.jl/tarball/d3d8fdeaea37bcae05191cad50a326e608035d97
Connecting to localhost:7890... connected.
Proxy request sent, awaiting response... 302 Found
Location: https://codeload.github.com/JuliaLang/JuliaSyntax.jl/legacy.tar.gz/d3d8fdeaea37bcae05191cad50a326e608035d97 [following]
--2022-08-19 15:44:47--  https://codeload.github.com/JuliaLang/JuliaSyntax.jl/legacy.tar.gz/d3d8fdeaea37bcae05191cad50a326e608035d97
Connecting to localhost:7890... connected.
Proxy request sent, awaiting response... 200 OK
Length: unspecified [application/x-gzip]
Saving to: ‘JuliaSyntax-d3d8fdeaea37bcae05191cad50a326e608035d97.tar.gz’

JuliaSyntax-d3d8fdeaea37bc     [  <=>                                  ] 122.64K   429KB/s    in 0.3s

2022-08-19 15:44:49 (429 KB/s) - ‘JuliaSyntax-d3d8fdeaea37bcae05191cad50a326e608035d97.tar.gz’ saved [125586]

woclass@wos-PC:~$ ll JuliaSyntax-d3d8fdeaea37bcae05191cad50a326e608035d97.tar.gz
-rw-r--r-- 1 woclass woclass 125586 Aug 19 15:44 JuliaSyntax-d3d8fdeaea37bcae05191cad50a326e608035d97.tar.gz
woclass@wos-PC:~$ md5sum JuliaSyntax-d3d8fdeaea37bcae05191cad50a326e608035d97.tar.gz
29ccd296e538403829ffd90b3cce67d8  JuliaSyntax-d3d8fdeaea37bcae05191cad50a326e608035d97.tar.gz
woclass@wos-PC:~$ sha512sum  JuliaSyntax-d3d8fdeaea37bcae05191cad50a326e608035d97.tar.gz
b3c26d6f7880f069b1f6f21ee3978cd3e797acd9ca6fb1e7ece28cf4dea576872bed3e8c3121d1d6cfe559518a17004595139f9978fd2591669cbe8128f7c1ac  JuliaSyntax-d3d8fdeaea37bcae05191cad50a326e608035d97.tar.gz
```